core/runtime/src/main/java/io/quarkus/runtime/logging/LoggingSetupRecorder.java

```diff
@@ -261,11 +259,9 @@ public static void initializeBuildTimeLogging(LogConfig config, LogBuildTimeConf
                 logCleanupFilter, launchMode);
 
         for (Map.Entry<String, CategoryConfig> entry : categories.entrySet()) {
-            final CategoryBuildTimeConfig buildCategory = isSubsetOf(entry.getKey(), buildConfig.categories);
-            final Level logLevel = getLogLevel(entry.getKey(), entry.getValue(), categories, buildConfig.minLevel);
-            final Level minLogLevel = buildCategory == null
-                    ? buildConfig.minLevel
-                    : buildCategory.minLevel.getLevel();
+            final String categoryName = entry.getKey();
+            final Level logLevel = getLogLevel(categoryName, categories, c -> c.level, buildConfig.minLevel);
+            final Level minLogLevel = getLogLevel(categoryName, buildConfig.categories, c -> c.minLevel, buildConfig.minLevel);
```

Comment:
So I added a few lambda expressions to runtime code (oh no), but the class had some before my change.

Alternatively, I could add getters to the `Config` classes and use method refs instead.